src/api/block_builder/input_selection/sender_issuer.rs

```diff
@@ -252,14 +199,190 @@ impl<'a> ClientBlockBuilder<'a> {
             }
         }
 
-        // Check required alias and nft outputs with new added outputs
-        // no need to check for sender and issuer again, since these outputs already exist and we don't set new features
-        // for them
+        // Check required Alias and NFT outputs with new added outputs.
+        // No need to check for sender and issuer again, since these outputs already exist and we don't set new features
+        // for them.
         let utxo_chain_inputs = self
             .get_utxo_chains_inputs(required_inputs.iter().map(|i| &i.output))
             .await?;
         required_inputs.extend(utxo_chain_inputs.into_iter());
 
-        Ok((force_use_all_inputs, required_inputs))
+        Ok(required_inputs)
+    }
+}
+
+// Select inputs for sender and issuer features
+pub(crate) fn select_inputs_for_sender_and_issuer<'a>(
+    inputs: impl Iterator<Item = &'a InputSigningData> + Clone,
+    selected_inputs: &mut Vec<InputSigningData>,
+    selected_inputs_output_ids: &mut HashSet<OutputId>,
```

Comment:
So this is supposed to be already filled with the IDs of `selected_inputs` ?

Reply:
Yes